### DISCUSSION TITLE
Building a Rust SDK with Example [ need to know how to integrate]

### DISCUSSION THREAD

**Discussion by mohanz** [modelcontextprotocol/modelcontextprotocol#1194] [Q&A - General] [answered]
### Pre-submission Checklist

- [X] This post follows the MCP community guidelines

### Questions
How do I add new sdk and example in rust if I have the repo read?

### What would you like to share?

# Rust version Model Context Protocol (MCP)
This is a Rust implementation of the Model Context Protocol (MCP) from @Anthropic.

## Minimalistic rust-idomatic implementation
Given it is still very early stage of MCP adoption, the goal of this is to provide a minimalistic working solution to start with.
Principle is to be flexible 

This implementation favors simplicity and ease of understanding to capture the core idea of MCP while maintaining compatibility with Claude Desktop.

## Comparison with other sdk

Comparing to the typescript sdk [typescript-sdk](https://github.com/modelcontextprotocol/typescript-sdk), this implementation avoid using callbacks and promises as much as possible.

For complete feature please refer to the [MCP specification](https://spec.modelcontextprotocol.io/).

## Why Rust?
In rust we trust.

## Features

- [x] Basic implementation of the MCP protocol
- [ ] Example servers
- [ ] Support for more transports (e.g. websocket)

**Comment by dsp-ant** [comment, marked as answer]
The best way to go about building an SDK is creating one and posting here for people to try. We assume that multiple SDKs will spin up in parallel and we want to see which shakes out to be the one working best for users before we would consider pulling it into the modelcontextprotocol organization. If you are interested in building a Rust SDK, I recommend creating a repository on your own account and starting there. Feel free to post in this discussion forum to attract peoples attention.

As a starting point, take a look at https://github.com/zed-industries/zed/tree/main/crates/context_server, which has a at least a nearly fully complete representation of the specification in types.rs

**Reply by mohanz** [reply]
thank you!

**Reply by mohanz** [reply]
https://github.com/AntigmaLabs/mcp-sdk

**Comment by localden** [comment]
We've limited Discussions in this repository to meeting notes and are closing this thread. To continue the conversation, pick a channel from [Contributor Communication](https://modelcontextprotocol.io/community/communication). Thanks for posting.
